HPXMLtoOpenStudio/resources/model.rb

```diff
@@ -1155,19 +1155,51 @@ def self.merge_unit_models(model, hpxml_osm_map)
       end
     end
 
-    model_objects.each do |obj|
+    # Create adjacent surfaces for HPXML surfaces w/ sameas attribute
+    merged_model_objects.each do |obj|
       next unless obj.to_Surface.is_initialized
-      next unless obj.to_Surface.get.additionalProperties.getFeatureAsString('hpxmlSameasID').is_initialized
 
-      surface_obj = obj.to_Surface.get
-      hpxml_sameas_id = surface_obj.additionalProperties.getFeatureAsString('hpxmlSameasID').to_s
-
-      adjacent_surface = model_objects.find { |o| o.to_Surface.is_initialized && o.to_Surface.get.additionalProperties.getFeatureAsString('hpxmlID').is_initialized && (hpxml_sameas_id == o.to_Surface.get.additionalProperties.getFeatureAsString('hpxmlID').to_s) }.to_Surface.get
-      surface_obj.setConstruction(adjacent_surface.construction.get.to_Construction.get.reverseConstruction)
-      adjacent_surface.setAdjacentSurface(surface_obj)
+      surface = obj.to_Surface.get
+      next unless surface.additionalProperties.getFeatureAsInteger('adjacentUnitNumber').is_initialized
+
+      adjacent_unit_number = surface.additionalProperties.getFeatureAsInteger('adjacentUnitNumber').get
+      adjacent_space_type = surface.additionalProperties.getFeatureAsString('adjacentSpaceType').get
+      adjacent_hpxml_id = make_variable_name(surface.additionalProperties.getFeatureAsString('adjacentHpxmlID').get, adjacent_unit_number)
+      adjacent_space_type = HPXML::LocationConditionedSpace if adjacent_space_type == HPXML::LocationBasementConditioned
+
+      unit_model = hpxml_osm_map.values[adjacent_unit_number]
+      unit_adjacent_space = unit_model.getThermalZones.find { |z| z.additionalProperties.getFeatureAsString('ObjectType').to_s == adjacent_space_type }.spaces[0]
+      adjacent_space = merged_model_objects.find { |s| s.to_Space.is_initialized && s.name.to_s == unit_adjacent_space.name.to_s }.to_Space.get
+
+      # Create/assign adjacent surface
+      adjacent_surface = surface.createAdjacentSurface(adjacent_space)
+      adjacent_surface = adjacent_surface.get
+      adjacent_surface.setName(adjacent_hpxml_id)
+
+      # Add new surface to unit_model as well so that the component loads program
+      # created later accounts for this surface.
+      unit_surface = OpenStudio::Model::Surface.new(adjacent_surface.vertices, unit_model)
+      surface_type = surface.additionalProperties.getFeatureAsString('SurfaceType').get
+      if surface_type == 'Floor'
+        surface_type = 'Ceiling'
+      elsif surface_type == 'Ceiling'
+        surface_type = 'Floor'
+      end
+      unit_surface.additionalProperties.setFeature('SurfaceType', surface_type)
+      unit_surface.setSpace(unit_adjacent_space)
+      unit_surface.setName(adjacent_surface.name.to_s)
```

Comment:
This was the final change I had to make to get the CI passing. Because we no longer create the adjacent surface during the unit model generation, it was being excluded from the component loads program (and so the component loads were not correctly summing to the total loads). I guess that is a disadvantage to this approach. But I still think this approach is a big improvement overall.

Reply:
I see, so this code is to add a corresponding surface to unit model with the same name, so that the sensor could be correctly added to the combined model. The component load program is fixed and CI is passing. It doesn't sound like there's impact on the component load program accuracy with the fix. 

It's a bit tricky that the component load program is still reading the unit model objects after model combination, I guess the reason is that we're storing a lot of information in the `additionalProperties` and it's only available at unit models? I think this is probably the best we can do and I'm fine with this workaround.